components/Search.js

```diff
@@ -14,54 +14,52 @@ class Search extends Component {
     id: PropTypes.string,
     searchButtonLabelText: PropTypes.string,
     layoutButtonLabelText: PropTypes.string,
+    onSearchCloseButtonClick: PropTypes.func
   };
 
   static defaultProps = {
     type: 'text',
     small: false,
     placeHolderText: '',
-    onChange: () => {},
+    onChange: () => {}
   };
 
   state = {
     format: 'list',
-    hasContent: this.props.value || this.props.defaultValue || false,
+    hasContent: this.props.value || this.props.defaultValue || false
   };
 
-  clearInput = () => {
-    if (this.props.value) {
-      this.props.onChange({
+  clearInput = evt => {
+    if (!this.props.value) {
+      this.input.value = '';
+      this.props.onChange(evt);
```

Comment:
@tw15egan and @marijohannessen pointed out that using `clearInput` was not firing `onChange` event in uncontrolled Search. So in this conditional that checks for controlled or uncontrolled, I added `this.props.onChange(evt)`. I'm not seeing duplicate `onChange` events being fired so I think this is ok. 

@chrisdhanaraj can you take on more look?